src/rust/multi.simplyhentai/src/lib.rs

```diff
@@ -0,0 +1,107 @@
+#![no_std]
+
+mod helper;
+mod parser;
+extern crate alloc;
+
+use aidoku::{
+	error::Result,
+	prelude::{format, get_chapter_list, get_manga_details, get_manga_list, get_page_list},
+	std::{
+		defaults::defaults_get,
+		net::{HttpMethod, Request},
+	},
+	Chapter, Filter, FilterType, Manga, MangaPageResult, Page,
+};
+use alloc::{string::String, vec::Vec};
+use helper::{make_search_url, API_BASE_URL, USER_AGENT};
+use parser::{parse_chapter_list, parse_manga, parse_page_list, parse_search};
+
+#[get_manga_list]
+fn get_manga_list(filters: Vec<Filter>, page: i32) -> Result<MangaPageResult> {
+	let mut manga_title: Option<String> = None;
+	let mut tags: Vec<String> = Vec::new();
+	let mut sort: Option<&str> = None;
+	let mut language = "English";
+	if let Ok(mut languages) = defaults_get("languages")?.as_array() {
```

Comment:
You're only handling one error here, which defeats the purpose. Don't return if `defaults_get` returns an error (don't use `?`).